Edit flow improvements

timwekkenbc opened the pull request:
- [x] Updated so that review branch removed on merge from github and "reviewBranch" status changed in db on update
- [x] Add new isPublished property to know if it's published
- [x] Added tests for ruleData service

timwekkenbc committed:
Updated so that review branch removed on merge from github and "reviewBranch" status changed in db on update, Add new isPublished property to know if it's published, Added tests for ruleData service

brysonjbest reviewed (approved):
Awesome changes for the management of rules, and the use of the published status is great. 

In the current implementation, deleted rules are only removed from the db, and not the repo. Would we want to explore adding a deletion of the repo file as a github commit requiring the standard approval process, so that deletions stay in sync between the front-end and rules repository? It looks like deleted rules currently will come back under the addUnsyncedFiles unless they are also removed from the repo?

timwekkenbc commented:
> Awesome changes for the management of rules, and the use of the published status is great.
> 
> In the current implementation, deleted rules are only removed from the db, and not the repo. Would we want to explore adding a deletion of the repo file as a github commit requiring the standard approval process, so that deletions stay in sync between the front-end and rules repository? It looks like deleted rules currently will come back under the addUnsyncedFiles unless they are also removed from the repo?

I think it's okay how it is for now. I don't expect the delete to be used too much beyond removing drafts that never made it to github. And I would want a delete to have to go through the same PR review process. But we'll see how things end up being used - might want to do that eventually.
